presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/SqlCustomization.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.jdbc;
+
+import java.util.List;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+
+/**
+ * Encapsulates strategies to use for various parts of SQL generation in {@link QueryBuilder}.
+ */
+public interface SqlCustomization
```

Comment:
`SqlDialect`?

Reply:
I think it's currently too simplistic to call it a dialect.
I don't think picking up "Dialect" name will future-proof us (so I wouldn't), but we may try.